### ISSUE TITLE
Model trains only on YOLO11n even after specifying model=yolov8n.pt

### ISSUE THREAD

**Issue by Jiya-N** [ultralytics/ultralytics#18572]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

I'm using `ultralytics==8.3.58` and `python==3.8.10`
Even after defining initialised weights to be `yolov8n.pt` or any weights other than `yolo11n.pt`, the YOLO function downloads `yolo11n.pt` and uses that for training the model
```
from ultralytics import YOLO
import torch 

device = torch.device(0 if torch.cuda.is_available() else ("mps" if torch.backends.mps.is_available() else "cpu"))
model_yolov8 = YOLO('yolov8n.pt')

model_yolov8.train(data="train.yaml", epochs=69, device=device)
```

`model_yolov8.train` downloads `yolo11n.pt` and uses that for training even though defined to use v8 weights

### Environment

.

### Minimal Reproducible Example

```
from ultralytics import YOLO
import torch 

device = torch.device(0 if torch.cuda.is_available() else ("mps" if torch.backends.mps.is_available() else "cpu"))
model_yolov8 = YOLO('yolov8n.pt')

model_yolov8.train(data="train.yaml", epochs=69, device=device)
```

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Jiya-N, thank you for bringing this to our attention and for your interest in Ultralytics 🚀! We recommend starting with the [Docs](https://docs.ultralytics.com) for detailed guidance on YOLO usage, including training with [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/).

If this is a 🐛 Bug Report, thank you for providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) — this helps us identify the issue more easily. An Ultralytics engineer will review this and assist you soon.

In the meantime, to rule out any version-related issues, please ensure you’ve upgraded to the latest `ultralytics` package along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

### Potential Troubleshooting

- Confirm that the `model` argument in `YOLO()` is pointing to the correct weights file (`yolov8n.pt` in this case).
- If the issue persists after upgrading, double-check your `train.yaml` configuration and any possible overrides on training parameters.

### Supported Environments

YOLO can be run in these official environments, all with preinstalled dependencies:
- **Notebooks**:
  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> 
  <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> 
  <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**:
  <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>. 
  See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/)

### Community and Support

Beyond GitHub, you can also engage with the Ultralytics community:
- Real-time discussions: Join us on [Discord](https://discord.com/invite/ultralytics) 🎧.
- Forums: Participate on [Discourse](https://community.ultralytics.com) for in-depth topics.
- Community threads: Share insights and questions on our [Subreddit](https://reddit.com/r/Ultralytics).

### Status

Check the current CI/CD pipeline status for Ultralytics below:
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

CI tests verify proper operation of YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu daily and on every code commit.

Thank you for your patience! An Ultralytics engineer will assist you shortly 😊

**Comment by Y-T-G** [comment]
> ### Search before asking
> 
> - [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.
> 

Duplicate of https://github.com/ultralytics/ultralytics/issues/17801

**Comment by visionNoob** [comment]
As mentioned by @Y-T-G   and #17807 

-> "It's downloaded for AMP checks, not for training."
That why `yolov11n.pt` is downloaded.
